src/components/Dropdowns/DropdownWithColorPicker.tsx

```diff
@@ -0,0 +1,77 @@
+import { Color as AntdColor } from "@rc-component/color-picker";
```

Comment:
Copied mostly from `DrawModeDropdown`, but removed some of the assumptions it was making about the data type (`DrawMode`). This component can now be reused for any option where a user may need to put in a custom color based on a selected mode.